client/payments/payment-recommendations.tsx

```diff
@@ -92,7 +110,14 @@ const PaymentRecommendations: React.FC = () => {
 				eventProps
 			);
 		}
-	}, [ shouldShowRecommendations, WcPayPromotionGateway, isLoading ] );
+	}, [ shouldShowRecommendations, WcPayPromotionGateway, isResolving ] );
+
+	useEffect( () => {
+		if ( ! installedPaymentGateway ) {
+			return;
+		}
+		window.location.href = installedPaymentGateway.settings_url;
```

Comment:
We previously were relying on the `setup-link` returned in the gateway response.  This is brittle since the settings page can be changed within the plugin.

The new check relies on fetching the gateway details after it has been installed and using its self-defined settings URL.

Resolvers in WP data (or at least the currently bundled version) do not return promises so we cannot await the payment gateway resolution.  I agree that it's slightly more confusing and I'm open to other pattern suggestions, but I believe this is the commonly used pattern.

On a side note, I think I recall someone mentioning that newer versions of WP data may allow us to await resolvers.